modules/navier_stokes/src/fv/actions/INSFVAction.C

```diff
@@ -0,0 +1,30 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "INSFVAction.h"
+#include "MooseApp.h"
+#include "TheWarehouse.h"
+#include "INSFVAttributes.h"
+
+registerMooseAction("NavierStokesApp", INSFVAction, "ns_meta_action");
+
+InputParameters
+INSFVAction::validParams()
+{
+  return Action::validParams();
+}
+
+INSFVAction::INSFVAction(InputParameters parameters) : Action(parameters) {}
+
+void
+INSFVAction::act()
+{
+  if (_current_task == "ns_meta_action")
+    _app.theWarehouse().registerAttribute<AttribINSFVBCs>("insfvbcs", 0);
```

Comment:
this catches internal sidesets for 
  [mesh]
    type = CartesianMeshGenerator
    dim = 2
    dx = '1'
    dy = '1 1 1'
    ix = '5'
    iy = '5 5 5'
    subdomain_id = '0
                    1
                    2'
  []
  [internal_boundary_bot]
    type = SideSetsBetweenSubdomainsGenerator
    input = mesh
    new_boundary = 'internal_bot'
    primary_block = 0
    paired_block = 1
  []
  [internal_boundary_top]
    type = SideSetsBetweenSubdomainsGenerator
    input = internal_boundary_bot
    new_boundary = 'internal_top'
    primary_block = 1
    paired_block = 2
  []

Reply:
the renaming the blocks is to avoid sharing the left / right sidesets among multiple blocks

Reply:
Is this a question or a statement? It will catch internal sidesets yes

Reply:
Ok so I should look at the part which deletes the internal sidesets from the check then.
This mesh leads to an error

Reply:
I can envision that it could error given the checks I've put in. What is the error?

Reply:
*** ERROR ***
Each boundary in an INSFV simulation must be covered by exactly one type of the following boundary condition types: INSFVFlowBC, INSFVNoSlipWallBC, INSFVSlipWallBC, INSFVSymmetryBC. However, the boundary 'internal_bot' is covered by 0 different types.

Reply:
I take it that the INSFV physics runs over all the subdomains? Yea it sounds like the problem is in `eraseInterfaces`. Do you want to share the full input file, and then I can debug it?

Reply:
Or if you have a solution you can push it here

Reply:
I dont have a solution yet.
There is only one subdomain once they are renamed. I sent you the full input file on slack, still looking at it though